src/lucky_flow/find_element.cr

```diff
@@ -13,27 +13,21 @@ class LuckyFlow::FindElement
   end
 
   def run
-    matching_elements.first? || retry
-  end
+    loop do
+      matching_elements = find_matching_elements
+      return matching_elements.first if matching_elements.first?
 
-  private def retry
-    self.tries += 1
-    if has_retries_left?
-      retry_after_delay
-    else
-      raise_element_not_found_error
+      break unless has_retries_left?
```

Comment:
By using `loop` with this break, we ensure it tries to find the element _at least_ once.

Reply:
Perfect 👍